src/components/settings/PushNotifications/hooks/useNotificationPreferences.ts

```diff
@@ -0,0 +1,217 @@
+  useEffect(() => {
+    hydrateUuidStore()
+  }, [hydrateUuidStore, uuidStore])
+
+  // Preferences state hydrator
+  const hydratePreferences = useCallback(() => {
+    if (!preferencesStore) {
+      return
+    }
+
+    getEntriesFromIndexedDb<PushNotificationPrefsKey, PushNotificationPreferences[PushNotificationPrefsKey]>(
+      preferencesStore,
+    )
+      .then((preferencesEntries) => {
+        setPreferences(Object.fromEntries(preferencesEntries))
+      })
+      .catch(() => null)
+  }, [preferencesStore])
+
+  // Hydrate preferences state
+  useEffect(() => {
+    hydratePreferences()
+  }, [hydratePreferences])
+
+  // Add store entry with default preferences for specified Safe(s)
+  const createPreferences = (safesToRegister: { [chain: string]: Array<string> }) => {
+    if (!preferencesStore) {
+      return
+    }
+
+    const defaultPreferencesEntries = Object.entries(safesToRegister).flatMap(([chainId, safeAddresses]) => {
+      return safeAddresses.map(
+        (safeAddress): [PushNotificationPrefsKey, PushNotificationPreferences[PushNotificationPrefsKey]] => {
+          const key = getPushNotificationPrefsKey(chainId, safeAddress)
+
+          const defaultPreferences: PushNotificationPreferences[PushNotificationPrefsKey] = {
+            chainId,
+            safeAddress,
+            preferences: _DEFAULT_NOTIFICATION_PREFERENCES,
+          }
+
+          return [key, defaultPreferences]
+        },
+      )
+    })
+
+    setManyIndexedDb(defaultPreferencesEntries, preferencesStore)
+      .then(hydratePreferences)
+      .catch(() => null)
+  }
+
+  // Update preferences for specified Safe
+  const updatePreferences = (
+    chainId: string,
+    safeAddress: string,
+    preferences: PushNotificationPreferences[PushNotificationPrefsKey]['preferences'],
+  ) => {
+    if (!preferencesStore) {
+      return
+    }
+
+    const key = getPushNotificationPrefsKey(chainId, safeAddress)
+
+    const newPreferences: PushNotificationPreferences[PushNotificationPrefsKey] = {
+      safeAddress,
+      chainId,
+      preferences,
+    }
+
+    setIndexedDb(key, newPreferences, preferencesStore)
+      .then(hydratePreferences)
+      .catch(() => null)
+  }
+
+  // Delete preferences store entry for specified Safe(s)
+  const deletePreferences = (safesToUnregister: { [chain: string]: Array<string> }) => {
+    if (!preferencesStore) {
+      return
+    }
+
+    const keysToDelete = Object.entries(safesToUnregister).flatMap(([chainId, safeAddresses]) => {
+      return safeAddresses.map((safeAddress) => getPushNotificationPrefsKey(chainId, safeAddress))
+    })
+
+    deleteManyFromIndexedDb(keysToDelete, preferencesStore)
+      .then(hydratePreferences)
+      .catch(() => null)
+  }
+
+  // Delete all preferences store entries
+  const clearPreferences = () => {
+    if (!preferencesStore) {
+      return
+    }
+
+    clearIndexedDb(preferencesStore)
+      .then(hydratePreferences)
+      .catch(() => null)
```

Comment:
We only need to catch errors, the returned value isn't needed. However, as we have the `no-empty` TSLint rule on, an "empty" function won't work. Would you prefer I change them to `try / catch`?

Reply:
I think the current style is elegant but I am not sure if we should log anything in case those indexed db calls throw. I will do some research on the topic.